modules/training/php/training.class.inc

```diff
@@ -115,19 +115,21 @@ class Training extends \NDB_Form
         // Get the instruments requiring certification from the config
         $certificationConfig      = $config->getSetting("Certification");
         $certificationInstruments = $certificationConfig['CertificationInstruments'];
-        $certificationInstruments['test']
-            = \Utility::toArray($certificationInstruments['test']);
         $instruments = array();
 
-        foreach ($certificationInstruments['test'] as $certificationInstrument) {
-            $test_key = $certificationInstrument['@']['value'];
-            $testID   = $DB->selectOne(
-                "SELECT ID
-                 FROM test_names
-                 WHERE Test_name
-                 LIKE '%$test_key%'"
-            );
-            $instruments[$testID] = $certificationInstrument['#'];
+        if (isset($certificationInstruments['test'])) {
+            $certificationInstruments['test']
+                = \Utility::toArray($certificationInstruments['test']);
+            foreach ($certificationInstruments['test'] as $certificationInstrument) {
+                $test_key = $certificationInstrument['@']['value'];
+                $testID   = $DB->selectOne(
+                    "SELECT ID
+             FROM test_names
+             WHERE Test_name
+             LIKE '%$test_key%'"
```

Comment:
these lines should be aligned with the query, other than that its ready to be approved